src/sugar3/activity/widgets.py

```diff
@@ -396,3 +396,31 @@ def __init__(self):
         self.paste = PasteButton()
         self.insert(self.paste, -1)
         self.paste.show()
+
+class CollabTextEditor(Gtk.Box):
+    def __init__(self,activity):
+        Gtk.Box.__init__(self)
+        self._collab = CollabWrapper(activity)
+        self._collab.connect('message', self.__message_cb)
+        self._collab.setup()
+        self.textview = Gtk.TextView()
+        self.textview.set_editable(True)
+        self.textview.set_cursor_visible(True)
+        self.textview.set_wrap_mode(Gtk.WrapMode.WORD)
+        self.textbuffer = self.textview.get_buffer()
+        self.textbuffer.connect('changed', self.__text_buffer_edited_cb)
+        self.textbuffer.set_text("TESTING")
+        self.textview.show()
+        self.pack_start(self.textview, expand=True, fill=True, padding=0)
+        self.show()
+
+    def __message_cb(self, collab, buddy, message):
+        action = message.get('action')
+	if action == 'entry_changed':
```

Comment:
On Sun, Dec 13, 2015, 10:06 PM Batchu Venkat Vishal <
notifications@github.com> wrote:

In src
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443076>
/
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443076>
sugar3
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443076>
/activity/
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443076>
widgets.py
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443076>
:

> + self._collab.connect('message', self.__message_cb) > +
self._collab.setup() > + self.textview = Gtk.TextView() > +
self.textview.set_editable(True) > + self.textview.set_cursor_visible(True)
> + self.textview.set_wrap_mode(Gtk.WrapMode.WORD) > + self.textbuffer =
self.textview.get_buffer() > + self.textbuffer.connect('changed',
self.__text_buffer_edited_cb) > + self.textbuffer.set_text("TESTING") > +
self.textview.show() > + self.pack_start(self.textview, expand=True,
fill=True, padding=0) > + self.show() > + > + def __message_cb(self,
collab, buddy, message): > + action = message.get('action') > + if action
== 'entry_changed':

You mean multiple text boxes should use the same collab setup? I'm a little
confused, could you elaborate a bit, thanks.

Yes.  So add another argument to the parameter that takes an ID for the
text box.  Then prefix all actions with that ID - so if I have 2 IDs they
don't sync.

—
Reply to this email directly or view it on GitHub
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274/files#r47443076>.<img
src="
https://ci6.googleusercontent.com/proxy/_aBzafILOCID-2ffcsDd82B8-k1Bvm5np_Fwcwa4UkKLA_k0WScmjkj5NJF-pmu3XuF2g0sBjCPUgfhBHHh8GfY4tJQDmc7UmDfqT-5EnVdjboGGZy4uexfzAySBFfz710aPCyO5iEPS-Rw7niDf5YYuWq4YoQ=s0-d-e1-ft#https://github.com/notifications/beacon/AFvjmhiXkyJsYxyQo8WqprCVhhwxfoGoks5pPUitgaJpZM4G0RrG.gif
">